src/transformers/trainer.py

```diff
@@ -3131,13 +3131,13 @@ def evaluation_loop(
 
             # Update containers on host
             if loss is not None:
-                losses = self.accelerator.gather_for_metrics((loss.repeat(batch_size)))
+                losses = self.gather_function((loss.repeat(batch_size)))
```

Comment:
This is only defined later right? line 3187 and 3858

Reply:
It's defined at 3846 yes, however when it's called should be well before any loops hit because it's during `create_accelerator_and_postprocess` which is done during `__init__`. Do you foresee an issue here with that @ArthurZucker ?

Reply:
As `create_accelerator_and_postprocess` is called in the `__init__` and a bunch of other attributes depend on this method being called in the init, I think this is OK.